tests-functional/clients/rpc.py

```diff
@@ -48,31 +45,19 @@ def verify_is_json_rpc_error(self, response):
         self._check_decode_and_key_errors_in_response(response, "error")
 
     @retry(stop=stop_after_delay(10), wait=wait_fixed(0.5), reraise=True)
-    def rpc_request(self, method, params=None, request_id=None, url=None, enable_logging=True):
-        if not request_id:
+    def rpc_valid_request(self, method, params=None, _id=None, url=None, enable_logging=True):
+        if not _id:
             request_id = self.request_counter
             self.request_counter += 1
+        else:
+            request_id = _id
+
         if params is None:
             params = []
         url = url if url else self.rpc_url
         data = {"jsonrpc": "2.0", "method": method, "id": request_id}
         if params:
             data["params"] = params
-        if enable_logging:
-            logging.debug(f"Sending POST request to url {url} with data: {json.dumps(data, sort_keys=True)}")
-        response = self.client.post(url, json=data)
-        try:
-            resp_json = response.json()
-            if enable_logging:
-                logging.debug(f"Got response: {json.dumps(resp_json, sort_keys=True)}")
-            if resp_json.get("error"):
-                assert "JSON-RPC client is unavailable" != resp_json["error"]
-        except JSONDecodeError:
-            if enable_logging:
-                logging.debug(f"Got response: {response.content}")
-        return response
-
-    def rpc_valid_request(self, method, params=None, _id=None, url=None, enable_logging=True):
-        response = self.rpc_request(method, params, _id, url, enable_logging=enable_logging)
-        self.verify_is_valid_json_rpc_response(response, _id)
+        response = self.api_valid_request("", data, url=url)
```

Comment:
Now when we are making RPC explicitly through `ApiClient`, I think it would be nice to make it an explicit call to `method="CallRPC"` here. Currently it's defined here, but doesn't make much sense there anymore:
https://github.com/status-im/status-go/blob/09bcb874fe136b1330466b5309e818c05ebd15e3/tests-functional/clients/status_backend.py#L59